lxd/daemon.go

```diff
@@ -1034,6 +1039,8 @@ func (d *Daemon) init() error {
 		NetworkAddress:       address,
 		ClusterAddress:       clusterAddress,
 		DebugAddress:         debugAddress,
+		MetricsAddress:       metricsAddress,
+		MetricsServer:        metricsServer(d),
```

Comment:
@monstermunchkin should this be only populated if the metrics address is being used, to avoid having the metrics server initialised when not in use?

Reply:
I personally don't see any problem having it initialized if not used. In theory we could add this check:
```go
if metricsAddress != "" {
    config.MetricsServer = metricsServer(d)
}
```

If `core.metrics_address` were to be unset again, we'd end up with the same issue we're trying to fix here unless we set `MetricsServer` to `nil`. By looking at the code that would require even more changes which IMHO are unnecessary.

Reply:
Well the reason I'm thinking that is otherwise we're going to increase the baseline memory requirements of LXD for reasons that many users won't need.

I just wondered what our position is on initialising optional features for everyone by default, it seems like that policy would lead to a high minimal memory requirement if followed for all new features.

Reply:
This isn't a blocker, but I'd be interested to know what @stgraber position is on this for future considerations.

Reply:
I looked into this a bit and came up with a possible solution. Unfortunately, it causes an `initialization loop` :disappointed: So for now, I won't be changing this as it's not a blocker.

Reply:
OK thanks for looking into it.